app/models/action.rb

```diff
@@ -0,0 +1,15 @@
+class Action < ApplicationRecord
+  enum status: [:pending, :completed, :failed]
+
+  belongs_to :workflow
+  belongs_to :subject
+
+  def perform
+    effect.perform(workflow_id, subject_id)
```

Comment:
should we rescue here and set the action status to failed? otherwise i don't see how it can reach that state